crates/iota-data-ingestion-core/src/worker_pool.rs

```diff
@@ -12,17 +12,51 @@ use iota_metrics::spawn_monitored_task;
 use iota_types::{
     full_checkpoint_content::CheckpointData, messages_checkpoint::CheckpointSequenceNumber,
 };
-use tokio::sync::{mpsc, oneshot};
+use tokio::{sync::mpsc, task::JoinHandle};
+use tokio_util::sync::CancellationToken;
 use tracing::info;
 
 use crate::{Worker, executor::MAX_CHECKPOINTS_IN_PROGRESS};
 
+type TaskName = String;
+type WorkerID = usize;
+
 pub struct WorkerPool<W: Worker> {
     pub task_name: String,
     concurrency: usize,
     worker: Arc<W>,
 }
 
+/// Represents the possible message types a WorkerPool can comminicate with
```

Comment:
small typo :wink:  (commUnicate)

Reply:
Thank you, [fixed](https://github.com/iotaledger/iota/pull/4648/commits/2eda721b05e84526cb29cc4bb968b326dbb206da#diff-dfef1eeaea282ac0a41b60863611b2244808890824a073bf2264ca6fc775a624R30-R41)